src/Env.jsx

```diff
@@ -0,0 +1,96 @@
+import { FormHelper } from "cockpit-components-form-helper.jsx";
+import { Grid } from "@patternfly/react-core/dist/esm/layouts/Grid";
+import { TextInput } from "@patternfly/react-core/dist/esm/components/TextInput";
+import { TrashIcon } from '@patternfly/react-icons';
+import cockpit from 'cockpit';
+
+import * as utils from './util.js';
+
+const _ = cockpit.gettext;
+
+export function validateEnvVar(env, key) {
+    switch (key) {
+    case "envKey":
+        if (!env)
+            return _("Key must not be empty");
+        break;
+    case "envValue":
+        break;
+    default:
+        console.error(`Unknown key "${key}"`);
+    }
+}
+
+const handleEnvValue = (key, value, idx, onChange, additem, itemCount, companionField) => {
+    // Allow the input of KEY=VALUE separated value pairs for bulk import only if the other
+    // field is not empty.
+    if (value.includes('=') && !companionField) {
+        const parts = value.trim().split(" ");
+        let index = idx;
+        for (const part of parts) {
+            const [envKey, ...envVar] = part.split('=');
+            if (!envKey || !envVar) {
+                continue;
+            }
+
+            if (index !== idx) {
+                additem();
+            }
+            onChange(index, 'envKey', envKey);
+            onChange(index, 'envValue', envVar.join('='));
+            index++;
+        }
+    } else {
+        onChange(idx, key, value);
+    }
+};
+
+export const EnvVar = ({ id, item, onChange, idx, removeitem, additem, itemCount, validationFailed, onValidationChange }) =>
+    (
+        <Grid hasGutter id={id}>
+            <FormGroup className="pf-m-6-col-on-md"
+                id={id + "-key-group"}
+                label={_("Key")}
+                fieldId={id + "-key-address"}
+                isRequired
+            >
+                <TextInput id={id + "-key"}
+                       value={item.envKey || ''}
+                       validated={validationFailed?.envKey ? "error" : "default"}
+                       onChange={(_event, value) => {
+                           utils.validationClear(validationFailed, "envKey", onValidationChange);
+                           utils.validationDebounce(() => onValidationChange({ ...validationFailed, envKey: validateEnvVar(value, "envKey") }));
+                           handleEnvValue('envKey', value, idx, onChange, additem, itemCount, item.envValue);
+                       }} />
+                <FormHelper helperTextInvalid={validationFailed?.envKey} />
+            </FormGroup>
+            <FormGroup className="pf-m-6-col-on-md"
+                id={id + "-value-group"}
+                label={_("Value")}
+                fieldId={id + "-value-address"}
+                isRequired
+            >
+                <TextInput id={id + "-value"}
+                       value={item.envValue || ''}
+                       validated={validationFailed?.envValue ? "error" : "default"}
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1354-20231018-104433-6efd18ea-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)

Reply:
Both error and default ought to be covered, perhaps it complains about the `?.`? Anyway, fine to ignore.

Reply:
"error" is not covered by tests. Because there is no error to test (see https://github.com/cockpit-project/cockpit-podman/pull/1354/files#diff-a55043e45dbe59249840f184fc8db573a6dc2294cb68a97d0c8797d1bb0282ceR20 ) that there is no validation for envValue. The code is here as a preparation for the future, once we expand validation futher